src/persist-client/src/lib.rs

```diff
@@ -282,10 +283,16 @@ impl PersistClient {
     /// If `shard_id` has never been used before, initializes a new shard and
     /// returns handles with `since` and `upper` frontiers set to initial values
     /// of `Antichain::from_elem(T::minimum())`.
+    ///
+    /// The given `blob_tags` are added as [S3 object tags] to all blobs written
+    /// on behalf of this WriteHandle.
+    ///
+    /// [S3 object tags]: https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-tagging.html
     #[instrument(level = "debug", skip_all, fields(shard = %shard_id))]
     pub async fn open<K, V, T, D>(
         &self,
         shard_id: ShardId,
+        blob_tags: HashMap<String, String>,
```

Comment:
maybe this could be a more generic container struct that initially just contains the blob tags, in case we discover other metadata we want to plumb through later

Reply:
yeah, I think that's reasonable. I'll wait to update until we resolve the plumbing question I posed and then do it all in one batch